### PR TITLE
Delete dead `Util.mkdirs`; inline use of `NoProxyEvaluator.shouldProxy`

### PR DESCRIPTION
**Pull request by jglick**
https://github.com/jenkinsci/remoting/pull/685#discussion_r1364311580

### CODE REVIEW

File: src/main/java/hudson/remoting/Util.java [hunk 1]
```diff
@@ -134,38 +138,9 @@ static URLConnection openURLConnection(URL url, String credentials, String proxy
             String encoding = Base64.getEncoder().encodeToString(proxyCredentials.getBytes(StandardCharsets.UTF_8));
             con.setRequestProperty("Proxy-Authorization", "Basic " + encoding);
         }
-        if (con instanceof HttpsURLConnection && sslSocketFactory != null) {
-            ((HttpsURLConnection) con).setSSLSocketFactory(sslSocketFactory);
-        }
```
Comment (jglick) [inline src/main/java/hudson/remoting/Util.java:139]: was not used

File: src/main/java/hudson/remoting/Util.java [hunk 2]
```diff
@@ -134,38 +138,9 @@ static URLConnection openURLConnection(URL url, String credentials, String proxy
...
-     */
-    static URLConnection openURLConnection(URL url) throws IOException {
-        return openURLConnection(url, null, null, null);
-    }
-
-    /**
-     * @deprecated Use {@link Files#createDirectories(java.nio.file.Path, java.nio.file.attribute.FileAttribute...)} instead.
-     */
-    @Deprecated
-    static void mkdirs(@NonNull File file) throws IOException {
```
Comment (jglick) [inline src/main/java/hudson/remoting/Util.java:164]: Unused.

File: src/main/java/org/jenkinsci/remoting/engine/JnlpAgentEndpointResolver.java [hunk 3]
```diff
@@ -481,7 +480,7 @@ else if(entry.endsWith("*"))
         }
         if (targetAddress == null) {
             String httpProxy = System.getenv("http_proxy");
-            if (httpProxy != null && !inNoProxyEnvVar(host)) {
+            if (httpProxy != null && NoProxyEvaluator.shouldProxy(host)) {
```
Comment (jglick) [inline src/main/java/org/jenkinsci/remoting/engine/JnlpAgentEndpointResolver.java:483]: inlining to simplify, and avoid use of negative sense

File: src/test/java/hudson/remoting/UtilTest.java [hunk 4]
```diff
@@ -1,78 +0,0 @@
...
-/**
- * @author Etienne Bec
- */
-public class UtilTest {
-
-    @Rule public TemporaryFolder temp = new TemporaryFolder();
-
-    @Test
-    @SuppressWarnings("deprecation")
-    public void mkdirs() throws IOException {
```
Comment (jglick) [inline src/test/java/hudson/remoting/UtilTest.java:45]: test of unused method

**Comment by basil** [comment]
Unlike #677, only deduplicates part of the duplicate code: the handling of `-cert` and `-noCertificateCheck`/`-disableHttpsCertValidation` (which, today, are only about 95% identical, not 100%) remains duplicated. In addition to being less comprehensive than #677, this PR actually inhibits the more comprehensive approach in #677 by creating needless merge conflicts due to arbitrary differences in refactoring choices.

**Review by basil — APPROVED** [review]

**Pull request merged**